content/posts/numpy/numpy2/index.md

```diff
@@ -0,0 +1,119 @@
+  updated, as proposed in [NEP 50](https://numpy.org/neps/nep-0050-scalar-promotion.html)
+  addressing surprising behaviors in type promotion e.g. with zero dimensional arrays.
+
+- Powerful new DType API and a new string dtype: NumPy 2.0 introduces a new API
+  for implementing user-defined custom data types as proposed by
+  [NEP 41](https://numpy.org/neps/nep-0041-improved-dtype-support.html). We used
+  this new API to implement `StringDType`, offering efficient and painless
+  support for variable length strings which was proposed in
+  [NEP 55](https://numpy.org/neps/nep-0055-string_dtype.html). And it is our hope
+  that enable future new data types with interesting new capabilities in the
+  PyData ecosystem and in NumPy itself.
+
+- Windows compatibility enhancements: The default 32-bit integer representation
+  on Windows has been updated to 64-bit on 64-bit architectures, addressing one
+  of the most common issues a developer runs into trying to get NumPy code to
+  work portably on Windows and Unix-like operating systems.
+
+- Full support for the Python array API standard: This is the first release to
+  include full support for the array API standard (v2022.12), which was enabled
+  by the promotion rules and API cleanup mentioned above, as well as by
+  adding new APIs and aligning existing APIs and behavior with the standard,
+  as proposed by [NEP 56](https://numpy.org/neps/nep-0056-array-api-main-namespace.html).
+
+These are just some of the more impactful changes in behavior and usability. In addition,
+NumPy 2.0 contains significant performance improvements, large documentation improvements,
+and more much - for a more extensive list of changes, see
+the [NumPy 2 release notes](https://numpy.org/devdocs/release/2.0.0-notes.html).
+
+A major release comes with changes that users may have to adapt to, but we
+worked hard to strike a balance between improvements and ensuring that the
+transition to NumPy 2.0 is as seamless as possible. We wrote a comprehensive
+[migration guide](https://numpy.org/devdocs/numpy_2_0_migration_guide.html),
+and a [ruff plugin](https://numpy.org/devdocs/numpy_2_0_migration_guide.html#ruff-plugin)
+helps to automatically update Python code so it will work with both NumPy 1.x and
+NumPy 2.x.
+
+While we do require C API users to recompile their projects to support running
+with NumPy 2.0, we prepared for this already in NumPy 1.25. The build process was
+simplified so that you can now always compile with the latest NumPy version.
+This means that projects build with NumPy 2.x are "magically" compatible with
+1.x. It also means that projects no longer need to build their binaries using
+the oldest version of NumPy supported by a project.
+
+We knew when we started the development for 2.0 that rolling out a NumPy 2.0
+will be (temporarily) disruptive, because of the backwards-incompatible API and
+ABI changes. We spent an extraordinary amount of effort communicating these
```

Comment:
```suggestion
ABI changes. We made a concerted effort to communicate these
```

Reply:
I'd prefer keeping the original phrasing, since I'd say it was a lot more than what "concerted effort" implies to me. We really went out of our way here and did a crazy amount of work on this - and not just by NumPy maintainers, but with other folks from for example RAPIDS and conda-forge joining in and making major testing, outreach and migration efforts proactively.

Reply:
That's totally fine; I'll resolve this then.

Reply:
Yeah, it's fair to say it was extraordinary.  It isn't typical that projects inform large set of downstram in issus explicitly :).